### ISSUE TITLE
`use-maxsplit-arg` false positive with custom split method

### ISSUE THREAD

**Issue by khameeteman** [PyCQA/pylint#4857]
### Bug description
I get a false positive when implementing a `split` method on a custom class:

```python
class Obj:

    def split(self, point):
        return 0, 1


obj = Obj()
obj.split((0, 0))[0]
```

Problem seems to be in the `[0]` part. When I change this to `[1]`, no warning is shown.

### Command used

```shell
pylint --rcfile=.pylintrc module
```

### Pylint output

```shell
C0207: Use obj.split((0, 0), maxsplit=1)[0] instead (use-maxsplit-arg)
```

### Expected behavior

No warning

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.6.14 (default, Aug  9 2021, 06:21:09)
```

**Comment by yushao2** [comment]
> /CC: @yushao2

take
